File: web/tests/reference/prefix-sum/subarray-sum-equals-k.js
function subarraySum(nums, k) {
  const seen = new Map([[0, 1]]);
  let running = 0;
  let count = 0;
  for (const x of nums) {
    running += x;
    count += seen.get(running - k) ?? 0;
    seen.set(running, (seen.get(running) ?? 0) + 1);
  }
  return count;
}

File: web/tests/reference/prefix-sum/subarray-sum-equals-k.py
def subarray_sum(nums, k):
    seen = {0: 1}
    running = 0
    count = 0
    for x in nums:
        running += x
        count += seen.get(running - k, 0)
        seen[running] = seen.get(running, 0) + 1
    return count
